core/trino-main/src/main/java/io/trino/execution/scheduler/policy/PhasedExecutionSchedule.java

```diff
@@ -215,15 +215,15 @@ private Set<PlanFragmentId> removeCompletedStage(StageExecution stage)
                 .filter(dependentFragmentId -> fragmentDependency.inDegreeOf(dependentFragmentId) == 1)
                 .collect(toImmutableSet());
         fragmentDependency.removeVertex(fragmentId);
-        fragmentTopology.removeVertex(fragmentId);
```

Comment:
this looks like separate fix, not strictly related. Separate commit?

Reply:
This is needed in this commit since we need to keep `fragmentTopology` even for non-scheduling stages